src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageTranslateDialog.vue

```diff
@@ -0,0 +1,295 @@
+			type: String,
+			required: true,
+		},
+		richParameters: {
+			type: Object,
+			required: true,
+		},
+	},
+
+	emits: ['close'],
+
+	setup() {
+		return { availableLanguages }
+	},
+
+	data() {
+		return {
+			isMounted: false,
+			selectedFrom: null,
+			selectedTo: null,
+			isLoading: false,
+			translatedMessage: '',
+		}
+	},
+
+	computed: {
+		userLanguage() {
+			return navigator.language.substring(0, 2)
+		},
+
+		sourceTree() {
+			const tree = {}
+			const uniqueSourceLanguages = Array.from(new Set(this.availableLanguages.map(element => element.from)))
+
+			uniqueSourceLanguages.forEach(language => {
+				tree[language] = {
+					id: language,
+					label: this.availableLanguages.find(element => element.from === language)?.fromLabel,
+					translations: this.availableLanguages.filter(element => element.from === language).map(model => ({
+						id: model.to,
+						label: model.toLabel,
+					})),
+				}
+			})
+
+			return tree
+		},
+
+		translationTree() {
+			const tree = {}
+			const uniqueTranslateLanguages = Array.from(new Set(this.availableLanguages.map(element => element.to)))
+
+			uniqueTranslateLanguages.forEach(language => {
+				tree[language] = {
+					id: language,
+					label: this.availableLanguages.find(element => element.to === language)?.toLabel,
+					sources: this.availableLanguages.filter(element => element.to === language).map(model => ({
+						id: model.from,
+						label: model.fromLabel,
+					})),
+				}
+			})
+
+			return tree
+		},
+
+		optionsFrom() {
+			return this.selectedTo?.id
+				? this.translationTree[this.selectedTo?.id]?.sources
+				: Object.values(this.sourceTree).map(model => ({
+					id: model.id,
+					label: model.label,
+				}))
+		},
+
+		optionsTo() {
+			return this.selectedFrom?.id
+				? this.sourceTree[this.selectedFrom?.id]?.translations
+				: Object.values(this.translationTree).map(model => ({
+					id: model.id,
+					label: model.label,
+				}))
+		},
+	},
+
+	watch: {
+		selectedTo() {
+			this.translatedMessage = ''
+		},
+		selectedFrom() {
+			this.translatedMessage = ''
+		},
+	},
+
+	mounted() {
+		this.selectedTo = this.optionsTo.find(language => language.id === this.userLanguage) || null
+
+		if (this.selectedTo) {
+			this.translateMessage()
+		}
+
+		this.$nextTick(() => {
+			// FIXME trick to avoid focusTrap() from activating on NcSelect
+			this.isMounted = !!this.$refs['translate-modal'].randId
+		})
+	},
+
+	methods: {
+		handleTranslate() {
+			this.translateMessage(this.selectedFrom?.id)
+		},
+
+		async translateMessage(sourceLanguage = null) {
+			try {
+				this.isLoading = true
+				const response = await translateText(this.message, sourceLanguage, this.selectedTo?.id)
+				this.translatedMessage = response.data.ocs.data.text
+			} catch (error) {
+				console.error(error)
+				showError(error.response?.data?.ocs?.data?.message ?? t('spreed', 'The message could not be translated'))
+			} finally {
+				this.isLoading = false
+			}
+		},
```

Comment:
On the first modal open it always fails to me from the request in `mounted`:

Request:

```js
fetch("https://nextcloud.dev.shgk.me/ocs/v2.php/translation/translate", {
  "headers": {
    "accept": "application/json, text/plain, */*",
    "accept-language": "en-US,en;q=0.9,ru;q=0.8",
    "content-type": "application/json",
    "requesttoken": "<DELETED>",
    "sec-ch-ua": "\"Chromium\";v=\"112\", \"Not:A-Brand\";v=\"99\"",
    "sec-ch-ua-mobile": "?0",
    "sec-ch-ua-platform": "\"Windows\"",
    "sec-fetch-dest": "empty",
    "sec-fetch-mode": "cors",
    "sec-fetch-site": "same-origin"
  },
  "referrerPolicy": "no-referrer",
  "body": "{\"text\":\"Hallo!\",\"fromLanguage\":null,\"toLanguage\":\"en\"}",
  "method": "POST",
  "mode": "cors",
  "credentials": "include"
});
```

Response:

```json
{"ocs":{"meta":{"status":"failure","statuscode":400,"message":""},"data":{"message":"Could not detect language"}}}
```

Reply:
It's on API side then, we always send text and 'to' language

Reply:
Yep, it seems, API is used correctly

Reply:
That's what I said in the office. You send a single word and fromLanguage is null.
But on a single word language models don't work and respond with a language, so we don't have a from when talking to translations provider and it returns with could not translate

Reply:
Maybe we should not show the error message in this case?
Otherwise, all small messages show an error toast on opening the translation modal.
Also, if only one language is available, we can pre-select it.

Reply:
Well we could try to hide the error from empty runs yeah, but let's do that in a follow up